src/runtime_src/core/tools/xbmgmt2/flash/xqspips.cpp

```diff
@@ -480,15 +480,14 @@ unsigned int XQSPIPS_Flasher::getFlashSize()
     return 0x8000000; //hard-code as 128MB so far
 }
 
-void XQSPIPS_Flasher::readBack(const std::string& output, unsigned int base)
+int XQSPIPS_Flasher::readBack(const std::string& output, unsigned int base)
 {
     initQSpiPS();
 
     uint32_t StatusReg = XQSpiPS_GetStatusReg();
-
     if (StatusReg == 0xFFFFFFFF) {
         std::cout << "[ERROR]: Read PCIe device return -1. Cannot get QSPI status." << std::endl;
-        return;
+        exit(-EOPNOTSUPP);
```

Comment:
is it exit? or return? same in all below case aswell

Reply:
Don't use exit. You can throw an exception in all these cases.

Reply:
Agree throw an exception.   Our codebase should have only 1 exit() call, which right after the entry point of this application (e.g., end of main())

Reply:
Noted.